sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/schema/BeamSqlRecord.java

```diff
@@ -17,24 +17,25 @@
  */
 package org.apache.beam.sdk.extensions.sql.schema;
 
-import com.google.auto.value.AutoValue;
-import java.io.Serializable;
 import java.util.List;
+import org.apache.beam.sdk.values.BeamRecord;
+import org.apache.beam.sdk.values.PCollection;
 
 /**
- * Field type information in {@link BeamSqlRow}.
- *
+ * {@link BeamSqlRecord} represents one row element in a {@link PCollection},
+ * with type provider {@link BeamSqlRecordTypeProvider}.
  */
-@AutoValue
-public abstract class BeamSqlRowType implements Serializable {
-  public abstract List<String> getFieldsName();
-  public abstract List<Integer> getFieldsType();
+public class BeamSqlRecord extends BeamRecord {
```

Comment:
`BeamSqlRecord` is a helper class for SQL, to avoid `(BeamSqlRowType) BeamRecord.getDataType();` everywhere. --In Beam SQL code, it heavily relies on `BeamSqlRecordTypeProvider` which has more functions than `BeamRecordTypeProvider `.